document/core/text/modules.rst

```diff
@@ -483,10 +493,20 @@ Element segments allow for an optional :ref:`table index <text-tableidx>` to ide
 .. math::
    \begin{array}{llclll}
    \production{element segment} & \Telem_I &::=&
-     \text{(}~\text{elem}~~\Tid^?~~x{:}\Ttableidx_I~~\text{(}~\text{offset}~~e{:}\Texpr_I~\text{)}~~y^\ast{:}\Tvec(\Tfuncidx_I)~\text{)} \\ &&& \qquad
-       \Rightarrow\quad \{ \ETABLE~x, \EOFFSET~e, \EINIT~y^\ast \} \\ &&|&
-     \text{(}~\text{elem}~~\Tid^?~~et{:}\Telemtype~~y^\ast{:}\Tvec(\Tfuncidx_I)~\text{)} \\ &&& \qquad
-       \Rightarrow\quad \{ \ETYPE~et, \EINIT~y^\ast \} \\
+     \text{(}~\text{elem}~~\Tid^?~~e{:}\Toffset_I~~y^\ast{:}\Tvec(\Tfuncidx_I)~\text{)} \\ &&& \qquad
+       \Rightarrow\quad \{ \ETABLE~0, \EOFFSET~e, \ETYPE~\FUNCREF, \EINIT~y^\ast \} \\ &&|&
+     \text{(}~\text{elem}~~\Tid^?~~e{:}\Toffset_I~~\{et, init\}{:}\Tfunclist_I~\text{)} \\ &&& \qquad
+       \Rightarrow\quad \{ \ETABLE~0, \EOFFSET~e, \ETYPE~et, \EINIT~init \} \\ &&|&
+     \text{(}~\text{elem}~~\Tid^?~~\text{(}~\text{table}~~x{:}\Ttableidx_I ~\text{)}~~e{:}\Toffset_I~~\{et, init\}{:}\Tfunclist_I~\text{)} \\ &&& \qquad
```

Comment:
Notation for pattern-matching on records via their punning on their labels, like here, hasn't been defined, so I suggest having funclist simple return a pair `(et,y*)` instead.